---
title: "2022_03_28_Isabela_Firre_Gradients_F5_F6_WT_v_KO"
output:
  pdf_document: default
  html_document: default
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(tidyverse)
library(rstatix)
library(ggpubr)
library(dplyr)
library(purrr)
library(janitor)
library(ggplot2)
library(ggvenn)
library(VennDiagram)
library(venneuler)
library(eulerr)
library(reshape)
library(tibble)
library(DESeq2)
knitr::opts_chunk$set(echo = TRUE)
```

#Making Tibbles of Spectral data for each protein from the Jan2022FA and Jan2022TVkit data
```{r}

#50ng protein
#Anthony said the 50ng yielded more genes identified
mass_spec_50ng_Mar2022_SucGrad <- read_csv("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/2022_03_24_50ng_protein_SCP_JRinn_PTN_MESC_results.csv") %>% clean_names()

Spectral_50ng <- mass_spec_50ng_Mar2022_SucGrad[,c(4,16,17,18,19)]


#100ng protein 
mass_spec_100ng_Mar2022_SucGrad <- read_csv("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/2022_03_24_100ng_protein_SCP_JRinn_PTN_MESC_results.csv") %>% clean_names()

Spectral_100ng <- mass_spec_100ng_Mar2022_SucGrad[,c(4,16,17,18,19)]

```

#Making Tibbles of Calculated Intensities for each dbp from the Jan2022FA and Jan2022TV data
```{r}
#50 ng protein 
Intensities_50ng <- mass_spec_50ng_Mar2022_SucGrad[,c(4,28,29,30,31)]
colnames(Intensities_50ng)[2:5] <- paste(colnames(Intensities_50ng)[2:5],"50ng",sep=".")


#100ng protein same thing. 
Intensities_100ng <- mass_spec_100ng_Mar2022_SucGrad[,c(4,28,29,30,31)]
colnames(Intensities_100ng)[2:5] <- paste(colnames(Intensities_100ng)[2:5],"100ng",sep=".")
```

#You'll see a shift to only analyzing the intensities. I can go back to Spectral Counts
#You'll also see that I'm favoring regular intensities right now. I'll go back and look at the other three types of intensities. Just want to get my bearing. 
#I'm going to filter out any genes with 0 intensities for WT and start treating the two technical runs as replicates
```{r}
#The plan is to separate out the WT genes
#Take a sum of each row
#Sort out those greater than 0
#Then grep those genes, grabbing both KO and WT at that point and also for both runs and putting them together in one dataframe to proceed forward. 
Intensities_50ng_WTcheck <- Intensities_50ng[,c(1,4,5)]
Intensities_100ng_WTcheck <- Intensities_100ng[,c(1,4,5)]
#hAVING PROBLEMS WITH RETURNING GENE NAMES AS NA IN THE BELOW FULL JOIN
Intensities_WTcheck <- full_join(Intensities_50ng_WTcheck,Intensities_100ng_WTcheck,by="gene",copy=FALSE,keep = FALSE)
Intensities_WTcheck <- as.data.frame(Intensities_WTcheck)
#resolving NA in values
Intensities_WTcheckA <- Intensities_WTcheck %>% replace_na(list(ptn_mesc_wt_p1_f5_f6_intensity.50ng = 0, ptn_mesc_wt_p2_f5_f6_intensity.50ng = 0, ptn_mesc_wt_p1_f5_f6_intensity.100ng=0, ptn_mesc_wt_p2_f5_f6_intensity.100ng=0))

#sum of each row
Intensities_WTcheck2 <- Intensities_WTcheckA %>% mutate(intensities_sum = rowSums(.[,sapply(., is.numeric)])) 
summary(Intensities_WTcheck2$intensities_sum)

Intensities_WTvalid <- dplyr::filter(Intensities_WTcheck2, intensities_sum > 0)
Intensities_WTvalid2 <- Intensities_WTvalid [1:(length(Intensities_WTvalid )-1)]

Intensities_valid1 <-  Intensities_WTvalid2 %>% left_join(Intensities_50ng[,c(1,2,3)],by="gene")
Intensities_valid <- Intensities_valid1 %>% left_join(Intensities_100ng[,c(1,2,3)],by="gene")
Intensities_valid <- Intensities_valid %>% replace_na(list(ptn_mesc_ko_p1_f5_f6_intensity.50ng = 0, ptn_mesc_ko_p2_f5_f6_intensity.50ng = 0, ptn_mesc_ko_p1_f5_f6_intensity.100ng=0, ptn_mesc_ko_p2_f5_f6_intensity.100ng=0))
```


```{r - prep table to run function}
Intensities_valid <- as.data.frame(t(Intensities_valid))
Intensities_newcol <- c("genotype","WT","WT","WT","WT","KO","KO","KO","KO")
Intensities_valid <- cbind(Intensities_newcol,Intensities_valid)
Intensities_valid <-Intensities_valid %>% row_to_names (row_number= 1) %>% clean_names()
Intensities_valid[] <- lapply(Intensities_valid, type.convert, as.is = TRUE)

```

#Wilcoxon Test on Intensities
#Output is a csv file to have the pvalues in a table format. 
```{r}

test.funi <- function(Intensities_valid, col) { 

 ci <- combn(unique(Intensities_valid$genotype),2)
 sigsi <- list()
 for(i in 1:ncol(ci)) {
    sigsi[[i]] <- wilcox.test(
                   Intensities_valid[Intensities_valid$genotype == ci[1,i],col],
                   Intensities_valid[Intensities_valid$genotype == ci[2,i],col], exact=FALSE
                 )
    }
    names(sigsi) <- paste("Group",ci[1,],"by Group",ci[2,])

 tests_Intensities_valid <- data.frame(Test=names(sigsi),
                    W=unlist(lapply(sigsi,function(x) x$statistic)),
                    p=unlist(lapply(sigsi,function(x) x$p.value)),row.names=NULL)

 return(tests_Intensities_valid)
}


tests_Intensities_valid <- lapply(colnames(Intensities_valid)[-1],function(x) test.funi(Intensities_valid,x))
names(tests_Intensities_valid) <- colnames(Intensities_valid)[-1]
```

#Writing CSV of the Wilcoxon Test
```{r}
results_Intensities_valid <- do.call(rbind.data.frame, tests_Intensities_valid)
write.csv(results_Intensities_valid,"/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//results_protein_Intensities.csv",row.names = TRUE)
```

#Histogram of p-values
```{r}
#intensities
hist(results_Intensities_valid$p, breaks = 20, main="Frequency of Wilcoxon Rank Sum Test P-values \nComparing KO with WT Mass Spec Intensities from SucGrad F5&6 Protein", xlab = "P-value from Wilcoxon Rank Sum Test", ylab = "Frequency of Genes", labels=TRUE) 

pdf("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//2022_03_28_Histogram_Wilcoxon_Intensities_protein_SucGrad_Mar2022.pdf")
hist(results_Intensities_valid$p, breaks = 20, main="Frequency of Wilcoxon Rank Sum Test P-values \nComparing KO with WT Mass Spec Intensitiesl Counts from SucGrad F5&6 Protein", xlab = "P-value from Wilcoxon Rank Sum Test", ylab = "Frequency of Genes", labels=TRUE) 
dev.off()

```

#Genes Below 0.05 and 0.10 p-value
#Feel free to change the names and the pvalue cutoffs
```{r}

significant <- c("very", "significant")
pattern_to_search <- paste0(significant, collapse = "|")


results_Intensities_valid <- results_Intensities_valid%>% 
  mutate(significance = ifelse(p < 0.05, "very",
                          ifelse(p < 0.10, "significant", "not")))

significantIntensities_valid_genes <- results_Intensities_valid[grepl(pattern_to_search, results_Intensities_valid$significance), ]

```

##will need to change the gene names below that I put in manually
##make sure genotype entry is the same as above
#Pvalues are on individual graphs, but not the big graph. 
#Graphs here are fancy and pretty close to publication ready. 
#Feel free to change the fill color. It's teal. 
```{r}
#genes with pvalues <0.05. Making Box Whisker Plots, both all together and individual

#Spectral
#in the future I might have to grep these names
sig_genes <- row.names(significantIntensities_valid_genes)
pattern_to_search_sig_genes <- paste0(sig_genes, collapse = "$|^")
pattern_to_search_sig_genes <- paste("^",pattern_to_search_sig_genes, "$", sep ="")
#pattern_to_search_valid_intensities <- paste0(valid_intensitiesgenes , collapse = "$|^")
#pattern_to_search_valid_intensities_ready <- paste("^",pattern_to_search_valid_intensities,"$",sep="")
#long_data_Spectral_sig_genes <- Spectral[, grep(pattern_to_search_sig_genes, colnames(Spectral))]
#genotype <- c("WT","WT","WT","WT","WT","WT","WT","WT","WT","WT","KO","KO","KO","KO","KO","KO","KO","KO","KO","KO")
#long_data_Spectral_sig_genes <- cbind(genotype,long_data_Spectral_sig_genes)
#mm_sigSpectral = melt(long_data_Spectral_sig_genes)

  #pdf("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_MassSpec_ChIRP_TVall//2022_03_28_sig_genes_Spectral_boxwhisker_Wilcoxon_S_MassSpec_Mar2022TV_v_Jan2022TV.pdf")
 # ggplot(mm_sigSpectral)+geom_boxplot(aes(x=genotype, y=value),fill = "#00AFBB")+facet_grid(.~variable)
  #dev.off()


#for (var in mm_sigSpectral$variable) { 
 
  #pdf(paste("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_MassSpec_ChIRP_TVall/2022_03_28_MassSpec_SpectralBoxPlots_SignificantGenes//", var, "_sig.pdf", sep=""))
 # print(ggpaired(mm_sigSpectral[mm_sigSpectral$variable==var,],x="genotype",y="value",fill="#00AFBB",line.color ="gray")+ggtitle(var)+stat_compare_means(label.x = 0.5))
  #dev.off()
 # }

#Intensities
long_data_Intensities_sig_genes <- Intensities_valid[, grep(pattern_to_search_sig_genes, colnames(Intensities_valid))]
genotype <- c("WT","WT","WT","WT","KO","KO","KO","KO")
long_data_Intensities_sig_genes <- cbind(genotype,long_data_Intensities_sig_genes)
mm_sigIntensities = melt(long_data_Intensities_sig_genes)

pdf("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//2022_03_30_sig_genes_Intensities_boxwhisker_Wilcoxon_MassSpec_protein.pdf")
ggplot(mm_sigIntensities)+geom_boxplot(aes(x=genotype, y=value),fill = "#00AFBB")+facet_grid(.~variable)
dev.off()

for (var in mm_sigIntensities$variable) { 
             
  pdf(paste("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/2022_03_30_Intensities_Protein_BoxWhiskerPlots//", var, "_sig.pdf", sep=""))
  #print(ggplot(mm_sigIntensities[mm_sigIntensities$variable==var,])+geom_boxplot(aes(x=genotype, y=value),fill="#00AFBB")+ggtitle(var))
  print(ggpaired(mm_sigIntensities[mm_sigIntensities$variable==var,],x="genotype",y="value",fill="#00AFBB",line.color ="gray")+ggtitle(var)+stat_compare_means(label.x = 0.5))
  dev.off()
}

#Trying to add pvalues to plots.
#Help websites
#https://www.datanovia.com/en/blog/ggpubr-how-to-add-p-values-generated-elsewhere-to-a-ggplot/
#https://www.r-bloggers.com/2017/06/add-p-values-and-significance-levels-to-ggplots/
#http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/
```

#Write a csv of significant genes and their pvalues (n=179)
```{r}
write.csv(significantIntensities_valid_genes,"/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//significant_genes_wilcoxonresults_protein_Intensities.csv",row.names = TRUE)
```

#I'm going to take the significant genes from Mar2022 and Jan_Mar2022
#And see how they fall in the SucGrad data
```{r}
pvalues_Mar2022_ChIRP <- read_csv("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_MassSpec_ChIRP_TVall/pvalues_Mar2022ChIRP_Siggenes.csv") %>% clean_names()
sig_intensity_Mar2022_genes <- pvalues_Mar2022_ChIRP$gene
pattern_to_search_sig_Mar2022_genes <- paste0(sig_intensity_Mar2022_genes, collapse = "$|^")
pattern_to_search_sig_Mar2022_genes <- paste("^",pattern_to_search_sig_Mar2022_genes, "$", sep ="")

sucgradintensities_Mar2022_sig_genes <- Intensities_valid[, grep(pattern_to_search_sig_Mar2022_genes, colnames(Intensities_valid))]
genotype_SucGrad <- c("WT","WT","WT","WT","KO","KO","KO","KO")
sucgradintensities_Mar2022_sig_genes <- cbind(genotype_SucGrad,sucgradintensities_Mar2022_sig_genes)
sucgradintensities_Mar2022_sig_genes2 <- as.data.frame(t(sucgradintensities_Mar2022_sig_genes))
colnames(pvalues_Mar2022_ChIRP)[3:4] <- paste(colnames(pvalues_Mar2022_ChIRP)[3:4],"Mar2022_ChIRP",sep=".")
#left_join(rownames_to_column(x.tsummary), sto.info, by=c("rowname" = "Symbol"))
colnames(results_Intensities_valid)[3:4] <- paste(colnames(results_Intensities_valid)[3:4],"Mar2022_SucGrad",sep=".")
sucgradintensities_Mar2022_sig_genes2 <- left_join(rownames_to_column(sucgradintensities_Mar2022_sig_genes2), pvalues_Mar2022_ChIRP, by=c("rowname" = "gene"))
#results<-merge(x=source1,y=source2,by=”State”,all.x=TRUE)
#sucgradintensities_Mar2022_sig_genes2 <- merge(x=sucgradintensities_Mar2022_sig_genes2, y=pvalues_Mar2022_ChIRP, all.x=TRUE)

sucgradintensities_Mar2022_sig_genes3 <- left_join(sucgradintensities_Mar2022_sig_genes2,rownames_to_column(results_Intensities_valid), by=c("rowname" = "rowname"))
#results<-merge(x=source1,y=source2,by=”State”,all.x=TRUE)
#sucgradintensities_Mar2022_sig_genes2 <- merge(x=sucgradintensities_Mar2022_sig_genes2, y=pvalues_Mar2022_ChIRP, all.x=TRUE)

write.csv(sucgradintensities_Mar2022_sig_genes3,"/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//sucgrad_intensities_Mar2022ChIRP_sig_genes_significant_genes_protein.csv",row.names = TRUE)

##Flip the table
##add pvalues from both SucGrad and Mar2022 ChIRP analysis

pvalues_Jan_Mar2022_ChIRP <- read_csv("/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_MassSpec_ChIRP_TVall/pvalues_Jan_Mar2022ChIRP_Siggenes.csv") %>% clean_names()
sig_intensity_Jan_Mar2022_genes <- pvalues_Jan_Mar2022_ChIRP$gene
pattern_to_search_sig_Jan_Mar2022_genes <- paste0(sig_intensity_Jan_Mar2022_genes, collapse = "$|^")
pattern_to_search_sig_Jan_Mar2022_genes <- paste("^",pattern_to_search_sig_Jan_Mar2022_genes, "$", sep ="")

sucgradintensities_Jan_Mar2022_sig_genes <- Intensities_valid[, grep(pattern_to_search_sig_Jan_Mar2022_genes, colnames(Intensities_valid))]
genotype_SucGrad <- c("WT","WT","WT","WT","KO","KO","KO","KO")
sucgradintensities_Jan_Mar2022_sig_genes <- cbind(genotype_SucGrad,sucgradintensities_Jan_Mar2022_sig_genes)
sucgradintensities_Jan_Mar2022_sig_genes2 <- as.data.frame(t(sucgradintensities_Jan_Mar2022_sig_genes))
colnames(pvalues_Jan_Mar2022_ChIRP)[3:4] <- paste(colnames(pvalues_Jan_Mar2022_ChIRP)[3:4],"Jan_Mar2022_ChIRP",sep=".")
#left_join(rownames_to_column(x.tsummary), sto.info, by=c("rowname" = "Symbol"))
sucgradintensities_Jan_Mar2022_sig_genes2 <- left_join(rownames_to_column(sucgradintensities_Jan_Mar2022_sig_genes2), pvalues_Jan_Mar2022_ChIRP, by=c("rowname" = "gene"))
#results<-merge(x=source1,y=source2,by=”State”,all.x=TRUE)
#sucgradintensities_Mar2022_sig_genes2 <- merge(x=sucgradintensities_Mar2022_sig_genes2, y=pvalues_Mar2022_ChIRP, all.x=TRUE)

sucgradintensities_Jan_Mar2022_sig_genes3 <- left_join(sucgradintensities_Jan_Mar2022_sig_genes2,rownames_to_column(results_Intensities_valid), by=c("rowname" = "rowname"))
#results<-merge(x=source1,y=source2,by=”State”,all.x=TRUE)
#sucgradintensities_Mar2022_sig_genes2 <- merge(x=sucgradintensities_Mar2022_sig_genes2, y=pvalues_Mar2022_ChIRP, all.x=TRUE)

write.csv(sucgradintensities_Jan_Mar2022_sig_genes3,"/scratch/Shares/rinn/tenaya/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO//sucgrad_intensities_Jan_Mar2022ChIRP_sig_genes_significant_genes_protein.csv",row.names = TRUE)
```


##Then comes histograms of the spectral and intensities counts
##Also more fancy box plots of the very interesting genes

##Isabela started working here##
```{r - merge raw intensities values for valid genes list}
#read in valid genes (179)
significantIntensities_valid_genes <- read.csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/significant_genes_wilcoxonresults_protein_Intensities.csv")

#merge intensities values
Intensities_50ng <- Intensities_50ng %>%
  mutate(ptn_name = tolower(Intensities_50ng$gene)) %>%
  select(!gene)

Intensities_100ng <- Intensities_100ng %>%
  mutate(ptn_name = tolower(Intensities_100ng$gene)) %>%
  select(!gene)

significantIntensities_valid_genes_values <- significantIntensities_valid_genes %>%
  dplyr :: rename(ptn_name = X) %>%
  left_join(Intensities_50ng) %>%
  left_join(Intensities_100ng)

#save this table
write_csv(significantIntensities_valid_genes_values, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sigInten_valid_genes_values.csv")
```

```{r - more strict filter for WT valid genes}
#I will consider for the analysis only genes that show intensities > 0 in both replicates
Intensities_50ng_WTcheck_filter <- Intensities_50ng_WTcheck %>%
  filter(!ptn_mesc_wt_p1_f5_f6_intensity.50ng == 0, !ptn_mesc_wt_p2_f5_f6_intensity.50ng == 0)

Intensities_100ng_WTcheck_filter <- Intensities_100ng_WTcheck %>%
  filter(!ptn_mesc_wt_p1_f5_f6_intensity.100ng == 0, !ptn_mesc_wt_p2_f5_f6_intensity.100ng == 0)

Intensities_WT_filter_both <- intersect(Intensities_50ng_WTcheck_filter$gene, Intensities_100ng_WTcheck_filter$gene)
Intensities_WT_filter_both <- as.data.frame(Intensities_WT_filter_both)
colnames(Intensities_WT_filter_both) <- "gene"
#list of 2320 genes to start comparison with KO
```

```{r - merge KO values and compare - 2320 genes}
Intensities_filter_both <- Intensities_WT_filter_both %>%
  left_join(Intensities_50ng) %>%
  left_join(Intensities_100ng)

#save this initial table 2320 genes + intensities values
write_csv(Intensities_filter_both, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/intensities_valid_filter.csv")

#prepare table#worked beautifully - but I'll keep Tenaya's format
#Intensities_filter <- Intensities_filter_both %>%
 # pivot_longer(
  # cols = 2:9,
  # names_to = c("condition", "replicate", "run"),
  # names_pattern = "ptn_mesc_?(.*)_p?(.*)_f5_f6_intensity.?(.*)",
  # values_to = "intensity")

#prepare table for function
Intensities_filter <- as.data.frame(t(Intensities_filter_both))
colnames(Intensities_filter) <- unlist(Intensities_filter[1, ], use.names = F)
Intensities_filter <- Intensities_filter[-c(1), ]
genotype <- c("KO","KO","WT","WT","KO","KO","WT","WT")
Intensities_filter <- Intensities_filter %>% add_column(genotype = genotype, .before = "Aar2")

Intensities_filter[] <- lapply(Intensities_filter, type.convert, as.is = TRUE)
Intensities_valid <- as.data.frame(Intensities_filter) #I had to keep the name in order to function work

#run function
#Wilcoxon Test on Intensities
#Output is a csv file to have the pvalues in a table format. 

test.funi <- function(Intensities_valid, col) { 

 ci <- combn(unique(Intensities_valid$genotype),2)
 sigsi <- list()
 for(i in 1:ncol(ci)) {
    sigsi[[i]] <- wilcox.test(
                   Intensities_valid[Intensities_valid$genotype == ci[1,i],col],
                   Intensities_valid[Intensities_valid$genotype == ci[2,i],col], exact=FALSE
                 )
    }
    names(sigsi) <- paste("Group",ci[1,],"by Group",ci[2,])

 tests_Intensities_valid <- data.frame(Test=names(sigsi),
                    W=unlist(lapply(sigsi,function(x) x$statistic)),
                    p=unlist(lapply(sigsi,function(x) x$p.value)),row.names=NULL)

 return(tests_Intensities_valid)
}


tests_Intensities_valid <- lapply(colnames(Intensities_valid)[-1],function(x) test.funi(Intensities_valid,x))
names(tests_Intensities_valid) <- colnames(Intensities_valid)[-1]

#write results
results_Intensities_valid <- do.call(rbind.data.frame, tests_Intensities_valid)
write.csv(results_Intensities_valid,"/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/results_protein_Intensities_filter.csv",row.names = TRUE)

#plot histogram of results
hist(results_Intensities_valid$p, breaks = 20, main="Frequency of Wilcoxon Rank Sum Test P-values \nComparing KO with WT Mass Spec Intensities from SucGrad F5&6 Protein - NEW FILTER", xlab = "P-value from Wilcoxon Rank Sum Test", ylab = "Frequency of Genes", labels=TRUE) 

pdf("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/2022_04_05_Histogram_Wilcoxon_Intensities_protein_SucGrad_filterWT_ITP.pdf")
hist(results_Intensities_valid$p, breaks = 20, main="Frequency of Wilcoxon Rank Sum Test P-values \nComparing KO with WT Mass Spec Intensities from SucGrad F5&6 Protein - NEW FILTER", xlab = "P-value from Wilcoxon Rank Sum Test", ylab = "Frequency of Genes", labels=TRUE) 
dev.off()

results_Intensities_valid_filter <- results_Intensities_valid 

#filter by pval - by Tenaya's code
significant <- c("very", "significant")
pattern_to_search <- paste0(significant, collapse = "|")

results_Intensities_valid_filter <- results_Intensities_valid_filter %>% 
  mutate(significance = ifelse(p < 0.05, "very",
                          ifelse(p < 0.10, "significant", "not")))

significantIntensities_valid_genes_filter <- results_Intensities_valid_filter[grepl(pattern_to_search, results_Intensities_valid_filter$significance), ]

#filter by pval value
sig_Int_genes_filter <- results_Intensities_valid_filter %>%
  filter(p <= 0.05)
#84 genes

#merge intensities values in the same table
sig_Int_genes_filter <- sig_Int_genes_filter %>%
  rownames_to_column(var = "gene") %>%
  left_join(Intensities_50ng) %>%
  left_join(Intensities_100ng)

#save this table = filter by WT (4 out of 4) and pvalfrom wilcox
write_csv(sig_Int_genes_filter, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sigInten_valid_genes_filter_values.csv")
```

I noticed that some genes have differences as higher intensities in KO - so let's filter only KO < WT
```{r - KO > WT (no intesity filter applied here)}
sig_Int_genes_filter_5 <- sig_Int_genes_filter %>%
  mutate(sum50_ko = rowSums(sig_Int_genes_filter[ , c(6,7)])) %>%
  mutate(sum50_wt = rowSums(sig_Int_genes_filter[ , c(8,9)])) %>%
  mutate(sum100_ko = rowSums(sig_Int_genes_filter[ , c(10,11)])) %>%
  mutate(sum100_wt = rowSums(sig_Int_genes_filter[ , c(12,13)]))

sig_Int_genes_filter_5 <- sig_Int_genes_filter_5 %>%
  filter(sum50_ko < sum50_wt & sum100_ko < sum100_wt)
#60 genes

#make this filter in Tenaya's first list - also keep only pval 0.05
#read in table with intensities values that I saved
significantIntensities_valid_genes_values <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sigInten_valid_genes_values.csv")

significantIntensities_valid_genes_doublefilter <- significantIntensities_valid_genes_values %>%
  mutate(sum50_ko = rowSums(significantIntensities_valid_genes_values[ , c(6,7)])) %>%
  mutate(sum50_wt = rowSums(significantIntensities_valid_genes_values[ , c(8,9)])) %>%
  mutate(sum100_ko = rowSums(significantIntensities_valid_genes_values[ , c(10,11)])) %>%
  mutate(sum100_wt = rowSums(significantIntensities_valid_genes_values[ , c(12,13)]))
  
significantIntensities_valid_genes_doublefilter <- significantIntensities_valid_genes_doublefilter %>% filter(significance == "very")

significantIntensities_valid_genes_doublefilter <- significantIntensities_valid_genes_doublefilter %>%
  filter(sum50_ko < sum50_wt & sum100_ko < sum100_wt)
#60 genes with good filtering

##WE GET THE SAME 60 GENES - independently of way of filtering WT in the initial list
#pval don't chance depending on initial list for wilcox, so after filter pval, we got the same list
```


```{r - filtering further what I found - out of 60 genes}
#now I'll filter ways of intensities = 0 after ko < wt filter

#intensity = 0 in at least one of KO replicates in any run
sig_Int_genes_filter_5.1 <- sig_Int_genes_filter_5 %>%
  filter(ptn_mesc_ko_p1_f5_f6_intensity.50ng == 0 | ptn_mesc_ko_p2_f5_f6_intensity.50ng == 0 | ptn_mesc_ko_p1_f5_f6_intensity.100ng == 0 | ptn_mesc_ko_p2_f5_f6_intensity.100ng == 0)
#21 genes

#now filtering considering occurence in both replicates for the same run
sig_Int_genes_filter_5.2 <- sig_Int_genes_filter_5 %>%
 filter(ptn_mesc_ko_p1_f5_f6_intensity.50ng == 0 & ptn_mesc_ko_p2_f5_f6_intensity.50ng == 0 | ptn_mesc_ko_p1_f5_f6_intensity.100ng == 0 & ptn_mesc_ko_p2_f5_f6_intensity.100ng == 0)
#14 genes

#save this as my top hits for now
write_csv(sig_Int_genes_filter_5.2, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sig_Int_genes_filter_5.2.csv")

#now filtering intensity = 0 in all KO
sig_Int_genes_filter_5.3 <- sig_Int_genes_filter_5 %>%
  filter(ptn_mesc_ko_p1_f5_f6_intensity.50ng == 0 & ptn_mesc_ko_p2_f5_f6_intensity.50ng == 0 & ptn_mesc_ko_p1_f5_f6_intensity.100ng == 0 & ptn_mesc_ko_p2_f5_f6_intensity.100ng == 0)
#4
```

##these filter might be applied before Wilcox test or after, it doesn't matter, results are the same

```{r - RANK INTENSITY}
#calculate fold change in the intensities and rank them
#no intensity = 0 applied here
#starting with 60 genes with pval < 0.05

sig_Int_genes_filter_5 <- sig_Int_genes_filter_5 %>%
  mutate(mean50_ko = rowSums(sig_Int_genes_filter_5[ , c(6,7)])) %>%
  mutate(mean50_wt = rowSums(sig_Int_genes_filter_5[ , c(8,9)])) %>%
  mutate(mean100_ko = rowSums(sig_Int_genes_filter_5[ , c(10,11)])) %>%
  mutate(mean100_wt = rowSums(sig_Int_genes_filter_5[ , c(12,13)])) 

sig_Int_genes_filter_5 <- sig_Int_genes_filter_5 %>%
  mutate(ratio_50 = sig_Int_genes_filter_5$mean50_wt/sig_Int_genes_filter_5$mean50_ko) %>%
  mutate(ratio_100 = sig_Int_genes_filter_5$mean100_wt/sig_Int_genes_filter_5$mean100_ko)

#very stringent filter
sig_Int_genes_filter_5.4 <- sig_Int_genes_filter_5 %>%
  filter(!ratio_50 < 1.5 & !ratio_100 < 1.5)
#31 genes

#save this as my second top hits for now
write_csv(sig_Int_genes_filter_5.4, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sig_Int_genes_filter_5.4.csv")

#see intersect between top hits #1 and top hits #2
intersect2 <- intersect(sig_Int_genes_filter_5.2$gene, sig_Int_genes_filter_5.4$gene)
#12 genes = TOP TOP hits
```

#TODO: filter or compare at some point the list with ko x wt RNA-seq data


#Another strategy would be use DESeq to calculate logFC and pvals
```{r - DESeq analysis}
#start working with the filtered list by WT occurency = 2320 genes

#read in if necessary
Intensities_filter_both <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/intensities_valid_filter.csv")

#1)organize samplesheet (create a table with sample information)
samplesheet_grad <- as.data.frame(colnames(Intensities_filter_both))

samplesheet_grad <- samplesheet_grad %>%
  dplyr::rename (sample_id = "colnames(Intensities_filter_both)") %>%
  filter(!sample_id == "gene")
samplesheet_grad$sample_id <- as.character(samplesheet_grad$sample_id)

genotype <- c("KO","KO","WT","WT","KO","KO","WT","WT")
replicate <- c("p1", "p2", "p1", "p2", "p1", "p2", "p1", "p2")
ptn_run <- c("50ng", "50ng", "50ng", "50ng", "100ng", "100ng", "100ng", "100ng")

samplesheet_grad <- samplesheet_grad %>% 
  add_column(genotype = genotype, replicate = replicate, ptn_run = ptn_run)

# Deseq wants the genotype (comparison) as factor so let's set that.
level <- c("KO", "WT")
samplesheet_grad$genotype <- factor(samplesheet_grad$genotype, 
                                         levels = level )
#save this if needed later
write_csv(samplesheet_grad, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/samplesheet_grad.csv")


#2)organize counts (intensity) table

####duplicated row names === 
#see the duplicated
Intensities_filter_both_dupli <- Intensities_filter_both[duplicated(Intensities_filter_both$gene), ]
#gene Tmpo
#I'll filter it out and change the names fot Tmpo.1/Tmpo.2/Tmpo.3 and rebind
gene_order <- c(".1", ".2", ".3")
Intensities_filter_both_dupli <- Intensities_filter_both_dupli %>%
  mutate(gene_new = paste(Intensities_filter_both_dupli$gene, gene_order)) %>%
  dplyr::select(!gene) %>%
  dplyr::rename (gene = gene_new)
colnames <- colnames(Intensities_filter_both)
Intensities_filter_both_dupli <- Intensities_filter_both_dupli[, colnames]

Intensities_filter_both <- Intensities_filter_both %>%
  filter(!gene == "Tmpo")

Intensities_filter_both_adj <- rbind(Intensities_filter_both, Intensities_filter_both_dupli)

counts <- Intensities_filter_both_adj %>%
  column_to_rownames(var = "gene")

# IMPORTANT: check cols in counts = samplesheet_grad$sample_id
all(colnames(counts) == samplesheet_grad$sample_id)

#save counts_intensity
write_rds(counts, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/counts_intensity.rds")
```

###WORKING HEREEEEEEE (nothing below is right)
```{r DESeq run}
#prepare to run DESeq

#change to matrix
counts <- as.matrix(counts)

# Now we need to round the count_extracts and change to integer from numeric.
counts <-  round(counts)

# change counts to integer (this is required by DESeq)
mode(counts) <- "integer"

##here to below
#run DESeq
deseq_run <- DESeqDataSetFromMatrix(counts, samplesheet_grad, 
                                      design = ~ genotype)

dds <- DESeq(deseq_run)
res1 <- results(dds)
res <- as.data.frame(res1)
#this table has the results as log2FC and pval

#I'll merge in this table the original intentity values (not the normalized values) and save
res_merged <- res %>%
  rownames_to_column(var = "gene") %>%
  left_join(Intensities_filter_both_adj)

write_csv(res_merged, "/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/results_deseq_intensities_grad.csv")

##these are normalized counts that can be used in many other analysis for now on - are the values used in deseq
rlog_counts <- deseq_run %>%
  rlog() %>%
  assay() %>%
  as.data.frame() %>% 
  rownames_to_column("gene")

rlog_counts <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/rlog_counts_deseq.csv")

```

Now let's filter the results

```{r - filter Deseq results}
#read in table if needed
res_merged <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/results_deseq_intensities_grad.csv")

res_filter <- res_merged %>%
  filter(padj <= 0.05, log2FoldChange >= 1.5)

#6 genes here, let's compare to what I got before

#this filter considered wilcox pval < 0.05, KO < WT, FC > 1.5 (simple ratio)
sig_Int_genes_filter_5.4 <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/results/sig_Int_genes_filter_5.4.csv")

intersect_deseq <- intersect(res_filter$gene, sig_Int_genes_filter_5.4$gene)
intersect_deseq <- as.data.frame(intersect_deseq)
#all included in both analysis
#TOP TOP TOP hits!!!

#now if we just consider logFC > 0
res_filter2 <- res_merged %>%
  filter(padj <= 0.05, log2FoldChange > 0)

intersect_deseq2 <- intersect(res_filter2$gene, sig_Int_genes_filter_5.4$gene)
intersect_deseq2 <- as.data.frame(intersect_deseq2)
```

Now let's compare with RNA-seq data from KO vs WT to be sure that the protein is low in the fractions because potentially binding to Firre and not because is not expressed anymore in the KO genotype at all

```{r - compare to RNA-seq data as a genotype background}
#Michael gave me the table of DEG in KO vs WT mESC
#read in
rna_seq <- read_csv("/scratch/Shares/rinn/isabela/ChChCh/2022_03_28_IsabelasGradientsFirreWTKO/firre_ko_vs_wt_mesc_deseq_results.csv")

#filter using usual parameters
rna_seq_filtered <- rna_seq %>%
  filter(padj <= 0.05, log2FoldChange <= -1.5)

rna_seq_up <- rna_seq %>%
  filter(padj <= 0.05, log2FoldChange >= 1.5)

rna_seq_up <- rna_seq %>%
  filter(log2FoldChange >= 1.5)
```
```

